src/audio_pixel.h

```diff
@@ -37,14 +40,50 @@ class audio_pixel_t {
 class audio_pixel_block_t {
 
 public: 
-    // TODO: handle the accessor not being valid
-    audio_pixel_block_t(MediaItem_Take* take, double pix_per_s) 
-        : m_take(take), m_pix_per_s(pix_per_s), 
-          m_accessor(safe_audio_accessor_t(take)) 
-          { /* TODO: */}
+    audio_pixel_block_t(double pix_per_s)
+        :m_pix_per_s(pix_per_s) {}
     ~audio_pixel_block_t() {}
 
-    void update() { /* TODO: see AudioAccessorStateChanged */}
+    void update(std::vector<double>& sample_buffer, int num_channels, int sample_rate) {
+        // constructs an audio pixel block using interleaved sample buffer and num channels
+        // clear the current set of audio pixels 
+        m_pixels.clear();
+
+        // vectors to store each channels audio pixel data
+        std::vector<double> channel_max(num_channels);
+        std::vector<double> channel_min(num_channels);
+        std::vector<double> channel_rms(num_channels);
+
+        // counter to help parse interleaved labels
+        int curr_channel = 0;
+
+        //calcualte the samples per audio pixel
+        int samples_per_pixel = (m_pix_per_s * sample_rate) * num_channels;
+        int current_sample_count = 1;
+
+        for (int i = 0; i < sample_buffer.size(); i++) {
```

Comment:
you could nest for loops to iterate through each channel individually. something like this

```cpp
for (int channel = 0 ; i < num_channels ; i++) {
  for (int i = channel ; i < sample_buffer.size(); i = i + num_channels) { 
    // do something for each channel
  }
}
```

Reply:
clever solution, thank you